geopandas/tests/test_crs.py

```diff
@@ -220,16 +220,24 @@ def test_dataframe(self):
         assert df.geometry.values.crs == self.osgb
 
         # different passed CRS than array CRS is ignored
-        with pytest.warns(FutureWarning):
+        crs_mismatch_msg = (
+            "CRS mismatch between CRS of the passed geometries "
+            r"and 'crs'. Use 'GeoDataFrame.set_crs\(crs, "
+            r"allow_override=True\)' to overwrite CRS or "
+            r"'GeoDataFrame.to_crs\(crs\)' to reproject geometries. "
+            "CRS mismatch will raise an error in the future versions "
+            "of GeoPandas."
+        )
+        with pytest.warns(FutureWarning, match=crs_mismatch_msg):
```

Comment:
```suggestion
        with pytest.warns(FutureWarning, match="CRS mismatch"):
```

The match doesn't need the whole warning string.